wiring/src/spark_wiring_ble.cpp

```diff
@@ -2177,8 +2168,11 @@ ssize_t BleLocalDevice::getAdvertisingData(BleAdvertisingData* advertisingData)
     if (advertisingData == nullptr) {
         return SYSTEM_ERROR_INVALID_ARGUMENT;
     }
+    hal_ble_adv_params_t advParams = {};
+    hal_ble_gap_get_advertising_parameters(&advParams, nullptr);
     advertisingData->clear();
-    size_t len = CHECK(hal_ble_gap_get_advertising_data(advertisingData->data(), BLE_MAX_ADV_DATA_LEN, nullptr));
+    advertisingData->resize((advParams.primary_phy == BLE_PHYS_CODED) ? BLE_MAX_SCAN_REPORT_BUF_LEN : BLE_MAX_ADV_DATA_LEN);
```

Comment:
This can fail to allocate and cause `hal_ble_gap_get_advertising_data()` to write out of bounds.

Reply:
Added check for resize failure